docs/proposals/20190709-cluster-spec-crds.md

```diff
@@ -0,0 +1,325 @@
+    |                 |                             |  |                         | |--+          |
+    |                 |                             |  |                         | |  | Provision|
+    |                 |                             |  |                         | |  | infra-   |
+    |                 | Set InfrastructureStatusRef |  |                         | |<-+ structure|
+    |                 |<----------------------------+--+-------------------------| |             |
+    |                 |                             |  |                         | |             |
+    |                 |                             |  +-----------------------------------------+
+    |                 |                             |                            +-+
+    |                 |  Cluster update             |                             |
+    |                 |---------------------------> |                             |
+    |                 |                            +-+                            |
+    |                 |  +-----------------------------------------+              |    
+    |                 |  | IF InfrastructureStatusRef              |              |    
+    |                 |  |    is set               | |             |              |    
+    |                 |  |                         | |             |              |    
+    |                 |  | Set InfrastructureStage | |             |              |    
+    |                 |  | to "Provisioned"        | |             |              |    
+    |                 |<-+-------------------------| |             |              |    
+    |                 |  |                         | |             |              |    
+    |                 |  +-----------------------------------------+              |
+    |                 |                            +-+                            |
+    |                 |                             |                             |
+```
+Figure 1: Cluster provisioning process
+
+Figure 1 presents the sequence of actions involved in provisioning a cluster, highlighting the coordination required between the CAPI cluster controller and the provider infrastructure controller. 
+
+The creation of the Cluster and provider infrastructure objects are independent events. When a Cluster object is created, if the `InfrastructureRef` is not set, the cluster controller will set the `InfrastructureState` to "Pending". When a provider infrastructure object is created, the provider's controller will do nothing unless its owner reference is set to a cluster. 
+
+When the `InfrastructureRef` is set in the cluster, the cluster controller will retrieve the infrastructure object. If the object has not been seen before, it will start watching it. Also, if the object's owner is not set, it will set to the Cluster object and will set the cluster's `InfrastructureState` to "Provisioning". 
```

Comment:
What if there are multiple clusters - how does the cluster controller know which one should own the infrastructure object?

Reply:
The cluster controller is handling the reconciliation of one cluster which has the reference to the cluster infrastructure. The cluster it is handling is the one which should own the cluster infrastructure.